qualtran/bloqs/arithmetic/sorting_test.py

```diff
@@ -27,19 +30,54 @@ def test_bitonic_sort(bloq_autotester):
 
 
 def test_comparator_manual():
-    bloq = Comparator(4)
-    assert bloq.t_complexity().t == 88 - 4
-    with pytest.raises(NotImplementedError):
-        bloq.decompose_bloq()
+    bloq = Comparator(2**4)
+    assert bloq.t_complexity().t == 88 - 4 - 7 * 4
+
+
+def test_comparator_symbolic_t_complexity():
+    bloq = _cmp_symb.make()
+    bitsize = ceil(log2(bloq.L - 1))
+    assert bloq.t_complexity() == TComplexity(t=15 * bitsize + 4, clifford=56 * bitsize + 24)
+
+
+@pytest.mark.parametrize("L", [5, 8, 12])
+def test_comparator_classical_sim(L: int):
+    bloq = Comparator(L)
+    for a in range(L):
+        for b in range(L):
+            res_a, res_b, anc = bloq.call_classically(a=a, b=b)
+            assert res_a <= res_b
```

Comment:
Also assert that `res_a` and `res_b` are equal to `min(a, b)` and `max(a,b)` and `anc` stores the result of the comparison.

Reply:
added the three assertions